native/cocos/bindings/jswrapper/v8/ScriptEngine.cpp

```diff
@@ -61,6 +61,18 @@ ccstd::unordered_map<ccstd::string, unsigned> jsbFunctionInvokedRecords;
         if (!(cond)) return val
 
 namespace se {
+AutoHandleScope::AutoHandleScope()
+: _handleScope(v8::Isolate::GetCurrent()) {
+#if CC_EDITOR
+    ScriptEngine::getInstance()->_getContext()->Enter();
+#endif
```

Comment:
This would change the defination of HandleScope

Reply:
The reason of this is to make sure the `context` is the one which we initializing engine. Cocos Creator Editor depends on Electron JS environment which may have more than one JS `context`. And while `Inspector Dev Tool` is attached, the current context may be a different one that will cause crash.

Yep, I agree that it seems that we modify the `Handle Scope` behavior a little bit, but I think it's currently the best place to fix that crash issue and it's wrapped in `CC_EDITOR` macro that will not affect `runtime` mode. Also binding code doesn't need to be changed. Besides, engine only uses one `context` now.